src/components/CheckForm/CheckForm.tsx

```diff
@@ -81,9 +81,13 @@ export const CheckForm = ({ check, disabled, pageTitle }: CheckFormProps) => {
   const styles = useStyles2(getStyles);
   const status = useCheckTypeOptions().find((option) => option.value === checkType)?.status;
   const isExistingCheck = Boolean(check);
-  const { isLoading, isOverCheckLimit, isOverScriptedLimit, isReady } = useLimits();
+  const { isLoading, isOverBrowserLimit, isOverHgExecutionLimit, isOverCheckLimit, isOverScriptedLimit, isReady } =
+    useLimits();
   const overLimit =
-    isOverCheckLimit || ([CheckType.MULTI_HTTP, CheckType.Scripted].includes(checkType) && isOverScriptedLimit);
+    isOverHgExecutionLimit ||
+    isOverCheckLimit ||
+    (checkType === CheckType.Browser && isOverBrowserLimit) ||
```

Comment:
Feels like the `useLimit` hook could return the `overlimit` value, which would make the code more readable and for the `overlimit` to have a single source of truth (also DRY).

`const { ..., isOverlimit } = useLimits(checkType)` 🤷🏻

Reply:
I explored this as I really liked this idea but I realised I struggled with it more than I'd like due to needing to know the reason in many cases why the limit had been reached to show the answer (is it the browser limit, the total check limit, execution count, etc.) so I am going to leave it as it is and we can circle back to improve this when we can.